lib/tbot/workloadidentity/workloadattest/kubernetes_unix.go

```diff
@@ -126,19 +128,80 @@ func (a *KubernetesAttestor) Attest(ctx context.Context, pid int) (*workloadiden
 	return att, nil
 }
 
-// getPodForID retrieves the pod information for the provided pod ID.
-// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/server/server.go#L371
-func (a *KubernetesAttestor) getPodForID(ctx context.Context, podID string) (*v1.Pod, error) {
+func (a *KubernetesAttestor) getPodAndContainerStatus(ctx context.Context, podID, containerID string) (*v1.Pod, *v1.ContainerStatus, error) {
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	log := a.log.With("pod_id", podID, "container_id", containerID)
+
+	retry, err := retryutils.NewRetryV2(retryutils.RetryV2Config{
+		Driver: retryutils.NewExponentialDriver(100 * time.Millisecond),
+		Max:    2 * time.Second,
+		Clock:  a.clock,
+	})
+	if err != nil {
+		return nil, nil, trace.Wrap(err, "creating retrier")
+	}
+
+	var (
+		pod             *v1.Pod
+		containerStatus *v1.ContainerStatus
+	)
+LOOP:
+	for {
+		pod, containerStatus, err = a.tryGetPodAndContainerStatus(ctx, podID, containerID)
+		switch {
+		case err != nil:
+			log.WarnContext(ctx, "Failed to get pod and container status from kubelet", "error", err)
```

Comment:
I'm not 100% sure about retrying on-error, because if we can't reach the attestor due to a misconfiguration we'll now block for 30s. It might be better to fail quickly.

WDYT?

Reply:
Hm - I'd definitely also be cautious about hammering the kubelet API - I can imagine a scenario shortly after a node comes online and there may be many workloads requesting an identity at once.

I'm definitely leaning towards propagating the failure.